backend/lib/components/Auth0.js

```diff
@@ -1,47 +1,69 @@
 const axios = require("axios");
-const {
-  config: { auth },
-} = require("../../config");
-const User = require("./User");
+const qs = require("querystring");
+const { config } = require("../../config");
 
-const oauth = async () => {
-  return axios({
-    method: "POST",
-    url: `${auth.host}/oauth/token`,
-    data: {
-      client_id: auth.clientId,
-      client_secret: auth.secretKey,
-      audience: `${auth.host}/api/v2/`,
-      grant_type: "client_credentials",
-    },
-  })
+const errorHandler = (err) => {
+  const { response } = err;
+  const { error, message } = response.data;
+  const statusCode = response.status || response.data.statusCode;
+  // eslint-disable-next-line no-throw-literal
+  throw { statusCode, error, message };
+};
+
+const buildOauthUrl = (provider) => {
+  const qParams = [
+    `scope=${provider.scope}`,
+    `response_type=code`,
+    `client_id=${config.auth.clientId}`,
+    `state=${config.auth.state}`,
+    `audience=${config.auth.domain}/api/v2/`,
+    `redirect_uri=${config.auth.appUrl}/login/callback`,
+    `connection=${provider.name}`,
+  ].join("&");
+  return `${config.auth.domain}/authorize?${qParams}`;
+};
+
+const authenticate = async (grantType, payload = {}) => {
+  const body = {
+    grant_type: grantType,
+    audience: `${config.auth.domain}/api/v2/`,
+    client_id: config.auth.clientId,
+    client_secret: process.env.AUTH_SECRET_KEY,
+    ...payload,
+  };
+  return axios
+    .post(`${config.auth.domain}/oauth/token`, qs.stringify(body), {
```

Comment:
Is it necessary to stringify the body for post? I think the object can be passed in directly.

Reply:
This is recommended by Auth0 documentation, do you have a reason for me to revert that?

Reply:
https://auth0.com/docs/api-auth/tutorials/authorization-code-grant#1-get-the-user-s-authorization

Reply:
Although I agree that we could remove it, I prefer to leave it as it is for now, because I tested a lot using this code and It's better to refactor after we add some tests